numpyro/infer/mixed_hmc.py

```diff
@@ -307,4 +307,6 @@ def body_fn(i, vals):
     def __getstate__(self):
         state = self.__dict__.copy()
         state["_wa_update"] = None
+        state["_prototype_trace"] = None
```

Comment:
could you also do this for the super class: DiscreteHMCGibbs?

Reply:
`_prototype_trace` is actually from `HMCGibbs` (parent of `DiscreteHMCGibbs`), so I've made the change there. I'm leaving the override of `__getstate__` in `MixedHMC`, as it requires additional attributes to be set to None compared to `HMCGibbs`.

I've tested that the issue previously occurred with `HMCGibbs`, `DiscreteHMCGibbs` and `HMCECS`, but should be fixed with the latest commit.